### PR TITLE
New division endpoint

### LINKED ISSUE #25: Endpoint to create an empty division

### PR DESCRIPTION
**Pull request by adamszcz**
# What I added (link any issues addressed)
Closes #25 
# How to test it
- Run ```https://localhost:5001/api/division/```
- Body:

```
{
    "Name": "Carrots"
}
```

- See that a ```Division``` object is returned and created in the database

### CODE REVIEW

File: CribblyBackend.UnitTests/ControllerTests/DivisionControllerTests.cs [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+            {
+                HttpContext = mockHttpContext.Object
+            };
+        }
+
+        [Fact]
+        public async Task Create_ShouldReturnOkAndCreatedId_IfNoError()
+        {
+            Division division = new Division();
+            mockDivisionService.Setup(x => x.Create(It.IsAny<Division>())).ReturnsAsync(division);
```
Comment (cszczepaniak) [inline CribblyBackend.UnitTests/ControllerTests/DivisionControllerTests.cs:41]: You might want to set this up to return a division with an ID so you can make sure the ID gets populated

**Review by cszczepaniak — APPROVED** [review]
Small comment on the test, but otherwise nice and straightforward!

File: CribblyBackend.UnitTests/ControllerTests/DivisionControllerTests.cs (hunk 1 above)
Comment (adamszcz) [reply CribblyBackend.UnitTests/ControllerTests/DivisionControllerTests.cs:41]: How does that look?

Comment (cszczepaniak) [reply CribblyBackend.UnitTests/ControllerTests/DivisionControllerTests.cs:41]:
```
var division = new Division
{
    Id = "gettin jiggy wit it",
};
mockDivisionService.Setup(...).ReturnsAsync(division);
var result = await divisionController.Create(new Division());
var typedResult = Assert.IsType<OkObjectResult>(result);
var div = Assert.IsType<Division>(typedResult.Value);
Assert.Equal(div.Id, "gettin jiggy wit it");

Comment (cszczepaniak) [reply CribblyBackend.UnitTests/ControllerTests/DivisionControllerTests.cs:41]: Now that I'm looking at that, though, this probably isn't the place to test for that -- you'd want to do something like this in the repository or service

Comment (adamszcz) [reply CribblyBackend.UnitTests/ControllerTests/DivisionControllerTests.cs:41]: Done!

**Pull request merged**